operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java

```diff
@@ -1376,7 +1376,11 @@ private Packet createPacket() {
     private class DomainPresenceInfoStep extends Step {
       @Override
       public NextAction apply(Packet packet) {
-        packet.with(DOMAINS.get(getNamespace()).get(getDomainUid()));
+        DomainPresenceInfo info =
+            Optional.ofNullable(DOMAINS.get(getNamespace())).map(n -> n.get(getDomainUid())).orElse(null);
+        if (info != null) {
+          info.addToPacket(packet);
+        }
```

Comment:
Since you already have an Optional, here, there is no need separately to test the resultant info. Instead, you can use `ifPresent` to control the addition of the info to the packet.